core/src/main/java/com/sequenceiq/cloudbreak/service/stack/flow/UpdateNodeCountValidator.java

```diff
@@ -71,6 +72,22 @@ public void validateHostGroupIsPresent(InstanceGroupAdjustmentV4Request instance
         }
     }
 
+    public void validataCMStatus(Stack stack, InstanceGroupAdjustmentV4Request instanceGroupAdjustmentJson) {
+        List<InstanceMetaData> instanceMetaDataAsList = stack.getInstanceMetaDataAsList();
+        List<InstanceMetaData> unhealthyCM = instanceMetaDataAsList.stream()
+                .filter(instanceMetaData -> (instanceMetaData.getInstanceMetadataType().equals(InstanceMetadataType.GATEWAY) ||
+                                                instanceMetaData.getInstanceMetadataType().equals(InstanceMetadataType.GATEWAY_PRIMARY))
+                                                && InstanceStatus.SERVICES_UNHEALTHY.equals(instanceMetaData.getInstanceStatus()))
+                .collect(Collectors.toList());
+        if (!unhealthyCM.isEmpty()) {
+            String notHealthyInstances = unhealthyCM.stream()
+                    .map(instanceMetaData -> instanceMetaData.getDiscoveryFQDN() + ": " + instanceMetaData.getInstanceStatus())
+                    .collect(Collectors.joining(","));
+            throw new BadRequestException(
+                    format("Upscale is not allowed because the CM host is not healthy: %s. Please repair!", notHealthyInstances));
```

Comment:
We should definitely not say "Please repair" - the CM instance being unhealthy could be caused by different reasons, including CCM connectivity. Do not want to encourage customers to click the 'Repair' button.

Also, should this be a specific check for SERVICES_UNHEALTHY ... or the reverse | must be SERVICES_HEALTHY state.